domain/entity/user.go

```diff
@@ -67,6 +80,7 @@ func (u *User) Validate() error {
 	return nil
 }
 
+//ValidatePassword
 func (u *User) ValidatePassword(p string) error {
 	err := bcrypt.CompareHashAndPassword([]byte(u.Password), []byte(p))
```

Comment:
I wonder whether bcrypt belongs in the domain layer. I mean the password hashing / validation should be, but `bcrypt` is a specific choice, that could be made in the application layer. Then you would inject a `PasswordSecurity` type into the domain layer, which contains the implementation details.

To determine what should be in the domain layer I refer to what a non-technical customer would say:

- "Yes, the pwds should be stored securely. Certainly not as plain text" 
- "Yes, the pwds should be hashed with bcrypt"

For small projects this is overkill, but DDD is usually overkill for small projects too, so yeah. Imagine you had a big SaaS project where a specific customer would say "We use our own password service". Then this functionality might even come via the infrastructure layer, but you have bcrypt hardwired in the domain.

Reply:
I completely agree with you, but i'm trying to keep the project as simple as possible. My goal is focus in the clean architecture and not in the details